logo-resize/lambda_function.py

```diff
@@ -0,0 +1,23 @@
+import boto3
+from PIL import Image
+
+s3_client = boto3.client('s3')
+
+def resize_image(image_path, resized_path):
+    with Image.open(image_path) as image:
+        image = image.resize((400, 400))
+        image.save(resized_path)
+
+def lambda_handler(event, context):
+    # get bucket name and file key from the event object
+    bucket_name = event['Records'][0]['s3']['bucket']['name']
+    key = event['Records'][0]['s3']['object']['key']
+    download_path = f'/tmp/{key}'
+    upload_path = f'/tmp/resized-{key}'
+    
+    if bucket_name == 'logos-raw':
+        new_bucket_name = 'logos-processed'
+        # download to /tmp directory, resize and upload to processed bucket
+        s3_client.download_file(bucket_name, key, download_path)
+        resize_image(download_path, upload_path)
+        s3_client.upload_file(upload_path, new_bucket_name, key)

```

Comment:
I am not sure how it is written in the task description.
But in general, resized and initial images could be stored in one bucket, just with different prefixes
* /logos-raw/logo_1.png
* /logos-processed/logo_2.png
As far as I remember, you can adjust trigger for a special `key` of the bucket

Reply:
I considered this option of returning to the same bucket, however AWS strongly discourages from doing this. This is depicted in the fact that when creating a trigger for the function you have to explicitly confirm that you understand that such actions may cause infinite loops and run-away Lambda functions (and consequently high costs). Also, I found the same strong advice against this in official AWS documentation, as well as some articles explaining how to create and use lambda functions. Also, in the project description it's not specified whether to use just one bucket, it's just said that the user should always get back the resized logo.
With all of this in mind, and after considering both options (single bucket vs two separate buckets), I decided to go with 2 buckets since it seems a safer option.